lib/active_model/serializer/association.rb

```diff
@@ -10,24 +10,86 @@ class Serializer
     #  Association.new(:comments, { serializer: CommentSummarySerializer })
     #
     class Association < Field
+      # TECHDEBT: remove unneeded accessors
```

Comment:
I'll do that if we decide to keep them.  Part of the issue is that we're passing through `options` but in some place it only needs one option or two, and in others, it mutates the options on e.g. the reflection.. so I'd like to make these more explicit and add sanity.. which make obviate the need for as many accessors as are now in this section.